packages/gluestick/src/renderer/helpers/linkAssets.js

```diff
@@ -26,69 +31,85 @@ const filterEntryName = (name: string): string => {
 };
 
 const getBundleName = ({ config }): string => {
-  const manifestFilename: string =
-    process.env.GS_VENDOR_MANIFEST_FILENAME || '';
+  const manifestFilename = process.env.GS_VENDOR_MANIFEST_FILENAME || '';
   const { buildDllPath } = config.GSConfig;
-  const manifestPath: string = path.join(
-    process.cwd(),
-    buildDllPath,
-    manifestFilename,
-  );
+  const manifestPath = path.join(process.cwd(), buildDllPath, manifestFilename);
   // Can't require it, because it will throw an error on server
   const { name } = JSON.parse(fs.readFileSync(manifestPath).toString());
-  // $FlowIgnore Server is compiled by webpack, so then we have access to webpack's public path
-  const publicPath: string = __webpack_public_path__ || '/assets/'; // eslint-disable-line camelcase,no-undef
+  const publicPath: string = __webpack_public_path__ || '/assets/'; // eslint-disable-line camelcase
   return `${publicPath}dlls/${name.replace('_', '-')}.dll.js`;
 };
 
-module.exports = function linkAssets(
+// Cache contents of CSS files to avoid hitting the filesystem
+const cache = {};
+
+const memoizedRead = async (name: string) => {
+  if (cache[name]) {
+    return cache[name];
+  }
+
+  const localPath = path.join(process.cwd(), 'build', 'assets', name);
+  const contents = await readFileAsync(localPath);
+
+  cache[name] = contents.toString();
+  return cache[name];
+};
+
+module.exports = async function linkAssets(
   { config }: Context,
   entryPoint: string,
   assets: Object,
   loadjsConfig: Object,
-): { styleTags: Object[], scriptTags: Object[] } {
-  const styleTags: Object[] = [];
-  const scriptTags: Object[] = [];
-  let key: number = 0;
-  const entryPointName: string = filterEntryName(entryPoint);
+) {
+  const styleTags = [];
```

Comment:
I removed a lot of unnecessary Flow typing from this file. All of this is inferred.